### PR TITLE
Problem: Internal failure state is not properly recognized

### PR DESCRIPTION
**Pull request by EldoreiJK**
Solution: Q&D fix added one more hardcoded state
@aquette you may want to provide better fix later

### CODE REVIEW

File: src/nut_agent.cc [hunk 1]
```diff
@@ -224,13 +224,21 @@ void NUTAgent::advertisePhysics ()
...
                 }
+                if (i == "Internal UPS fault!") {
+                    internal_failure_bit = bit;
+                }
                 bit++;
             }
+            // TODO FIXME I hate this kind of fix, but it was to be quick and dirty
+            if (alarms.find("Internal failure!") != std::string::npos) {
+                bitfield |= (1 << internal_failure_bit);
+            }
```
Comment (jimklimov) [inline src/nut_agent.cc]: For clarity to future reviewers, we have a LUA rule-file that actually uses specific bit numbers to decide on criticality of the failure. Here we "remember" the number in array of the "Internal UPS fault" message (above in the loop), and alias it to other string(s... later?) with equivalent status reports that we have seen over other NUT drivers and protocols (e.g. NetXML vs SNMP).

**Comment by jimklimov** [comment]
There are more proper fix approaches for this, but they are not so quick... and would impact numerous components, so need to be scheduled as a separate effort. For one part, the list of messages should likely be a loadable text file (or directory of files) to stand any chance of scaling to more devices and vendors and their typo's etc., and probably a separate list of severities or event-types, because we are likely to have similar cases of different ways to say the same thing later on.

File: src/nut_agent.cc (hunk 1 above)
Comment (EldoreiJK) [reply src/nut_agent.cc]: Better fix would be to remove bitfield, or to support vector of vectors for strings with the same meaning.

**Comment by EldoreiJK** [comment]
Long-term solution is indeed external file with definition of states, but that would also require alert rules to be able to load the content and parse it properly, which was not initial design of rules (as all the details required for such rules should be inside it's file). There would definitely need to be a discussion how to address this properly, taking into account project future and requirements.

**Pull request merged**